python/ray/tune/suggest/ax.py

```diff
@@ -190,20 +185,26 @@ def _setup_experiment(self):
                 objective_name=self._metric,
                 parameter_constraints=self._parameter_constraints,
                 outcome_constraints=self._outcome_constraints,
-                minimize=self._mode != "max")
+                minimize=self._mode != "max"
+            )
         else:
             if any([
                     self._space, self._parameter_constraints,
-                    self._outcome_constraints
+                    self._outcome_constraints, self._mode, self._metric
             ]):
                 raise ValueError(
                     "If you create the Ax experiment yourself, do not pass "
                     "values for these parameters to `AxSearch`: {}.".format([
-                        "space", "parameter_constraints", "outcome_constraints"
+                        "space", 
+                        "parameter_constraints", 
+                        "outcome_constraints",
+                        "mode",
+                        "metric",
                     ]))
 
         exp = self._ax.experiment
-        self._objective_name = exp.optimization_config.objective.metric.name
+        self._mode = "min" if exp.optimization_config.objective.minimize else "max"
```

Comment:
Added setting `self._mode` here, as otherwise the users would run into an error in `suggest` because metric mode is undefined, even if it was set on the Ax experiment

Reply:
That makes sense, thanks for catching this. This breaks some tests however, would you mind fixing those?